Gopkg.lock

```diff
@@ -0,0 +1,153 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
+
+
+[[projects]]
+  name = "github.com/Azure/go-autorest"
+  packages = ["autorest","autorest/adal","autorest/azure","autorest/date","autorest/to","autorest/validation"]
+  revision = "c67b24a8e30d876542a85022ebbdecf0e5a935e8"
+  version = "v9.4.1"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/MichaelTJones/walk"
+  packages = ["."]
+  revision = "4748e29d5718c2df4028a6543edf86fd8cc0f881"
+
+[[projects]]
+  name = "github.com/dgrijalva/jwt-go"
+  packages = ["."]
+  revision = "dbeaa9332f19a944acb5736b4456cfcc02140e29"
+  version = "v3.1.0"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/dnaeon/go-vcr"
+  packages = ["cassette","recorder"]
+  revision = "87d4990451a858cc210399285be976e63bc3c364"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/howeyc/gopass"
+  packages = ["."]
+  revision = "bf9dde6d0d2c004a008c27aaee91170c786f6db8"
+
+[[projects]]
+  name = "github.com/marstr/collection"
+  packages = ["."]
+  revision = "871b1cfa2ab97d3d8f54a034280907896190c346"
+  version = "v0.3.3"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/marstr/goalias"
+  packages = ["model"]
+  revision = "3026ca7e18e829805b990b41e4ce28c92cb80caa"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/marstr/randname"
+  packages = ["."]
+  revision = "3ef1f47af99e66171417047e6f6fa334345e95e7"
+
+[[projects]]
+  name = "github.com/mattn/go-colorable"
+  packages = ["."]
+  revision = "167de6bfdfba052fa6b2d3664c8f5272e23c9072"
+  version = "v0.0.9"
+
+[[projects]]
+  name = "github.com/mattn/go-isatty"
+  packages = ["."]
+  revision = "0360b2af4f38e8d38c7fce2a9f4e702702d73a39"
+  version = "v0.0.3"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/mgutz/ansi"
+  packages = ["."]
+  revision = "9520e82c474b0a04dd04f8a40959027271bab992"
+
+[[projects]]
+  branch = "master"
+  name = "github.com/mgutz/minimist"
+  packages = ["."]
+  revision = "39eb8cf573ca29344bd7d7e6ba4d7febdebd37a9"
+
+[[projects]]
+  name = "github.com/mgutz/str"
+  packages = ["."]
+  revision = "968bf66e3da857419e4f6e71b2d5c9ae95682dc4"
+  version = "v1.2.0"
+
+[[projects]]
+  name = "github.com/mgutz/to"
+  packages = ["."]
+  revision = "00c06406c2dd2e011f153a6502a21473676db33f"
+  version = "v1.0.0"
+
+[[projects]]
+  name = "github.com/nozzle/throttler"
+  packages = ["."]
+  revision = "d9b45f19996c645d38c9266d1f5cf1990e930119"
+  version = "v1.0"
+
+[[projects]]
+  name = "github.com/satori/go.uuid"
```

Comment:
from here https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md only one of the `version`, `revision` or `branch` can be specified

Reply:
Same for the other constraints

Reply:
or that is for the toml file and this will have everything ?

Reply:
You are correct, that is only for the `.toml`. For the `.lock` file, it is auto-generated by the tool and specifies everything it needs for context, including version info and commit ids.